### PR TITLE
Fixing vagrant issues on mac os x 

### LINKED ISSUE #4006: Error when accessing poll without being logged in
When I try to access a poll through the poll editor ([example](http://direct.codecombat.com/editor/poll/after-playing-codecombat-how-interested-are-you-in-programming)) without having access to it (like when I'm logged out or logged in with a normal account), I get this error in the console:
- Uncaught TypeError: Cannot read property 'build' of null @ PollEditView.coffee:58

Note: Similar to #3266.

### PR DESCRIPTION
**Pull request by dfberry**
1. fixes error for box not found by using correct version of vagrant
     otherwise, you have to use the hardcoded url for the box:
     config.vm.box_url = "https://oss-binaries.phusionpassenger.com/vagrant/boxes/latest/ubuntu-14.04-amd64-vbox.box"
2. fixes issue with most current vagrant
3. fixes issue with script permission error for applying dump (issue 4061)
4. fixes #4006

### CODE REVIEW

File: test/app/views/editor/poll/PollEditView.spec.coffee [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+      _id: '1234'
+      anonymous: false
+      email: 'some@email.com'
+      name: 'Existing User',
+      permissions: ['admin']
+    })
+
+    view = new PollEditView({}, @poll.id)
+    view.render()
+    console.log(view.$el.find('#modal-error'))
```
Comment (sderickson) [inline test/app/views/editor/poll/PollEditView.spec.coffee]: Extraneous comment

**Comment by sderickson** [comment]
Looks good! Please remove the extraneous logs, and then it'll be good to go.

File: test/app/views/editor/poll/PollEditView.spec.coffee (hunk 1 above)
Comment (dfberry) [inline test/app/views/editor/poll/PollEditView.spec.coffee]: I'm going to separate out the two issues in different PRs and remove the test - I couldn't get the test of noty to work as I wanted.
